packages/api/src/shared/helpers/DateHelper.ts

```diff
@@ -28,6 +28,20 @@ export const englishMonthNames = Object.values(frenchToEnglishMonthsMap).map(mon
     capitalizeFirstLetter(monthLowercase)
 );
 
+export function getMonthlyDataObject(dbData, index1Key, dataKey) {
```

Comment:
Je ne sais pas si DateHelper est le meilleur endroit pour avoir cette méthode.

Si je comprend bien l'idée c'est de récupérer réduire le résultat mongo qui est un objet d'objets ayant chacun un index1key (donc un monthId) et un dataKey (donc un cumulativeNbUsers) dans un object avec comme clés les mois et en valeur le cumul calculé par mongo.

Vu que c'est extrêmement lié à mongo je mettrai cette fonction dans un MongoQueryHelper ou un truc comme ça. Qu'en penses-tu ?

PS: je diviserai la méthode en deux avec un reduceObjectToKeyValue(obj, indexKey, valueKey) (puisque ton for ... of est au final un reduce) car cette méthode pourra être réutiliser plus tard sans avoir la notion de date !

Reply:
Laisser ici mais le faire en une seule passe + essayer de trouver meilleur nom

Reply:
J'ai fait d'autres modifs, mais en fait si on ne fait qu'une seule passe, 
- soit on parcourt `dbData` mais on ne crée pas les entrées pour les mois qui ne sont pas présents en BDD
- soit on parcourt `englishMonthNames` mais il faut faire des recherches dans `dbData` pour le bon index et on a une moins bonne complexité
Donc même si o fait deux passes c'est le plus optimal pour avoir ce résultat